feat(#56): 전적 관련 기능 구현

ohhamma opened the pull request:
## 📌 연관된 이슈
- #56
- #60 

## 📝 작업 내용
- 1v1 전적 조회 및 추가
- 토너먼트 전적 조회 및 추가
- 게임 db 중복 문제 해결

## 🌳 작업 브랜치명
- `feat/game`

## 📸 스크린샷 (선택)
<img width="1582" alt="스크린샷 2024-04-11 21 23 08" src="https://github.com/bitbit-Merry-go-round/Transcendence-BE/assets/81293930/21dc8746-9b20-429a-bfbe-19b8b915cbab">

## 💬 리뷰 요구사항 (선택)
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

ohhamma committed:
chore: delete db_init.sh

ohhamma committed:
chore: delete tests.py

hyemch committed:
feat: game 컨테이너 생성 및 django 프로젝트 생성 (#56)

hyemch committed:
feat: game 컨테이너 생성 및 django 프로젝트 생성 (#56)

ohhamma committed:
style: format backend code

ohhamma committed:
style: format json response

hyemch committed:
rename: 장고 앱 이름 변경 (#56)

ohhamma committed:
fix: handle user exceptions (#46)

ohhamma committed:
fix: handle logout exception (#46)

ohhamma committed:
style: format requests code

ohhamma committed:
fix: handle bearer token exception (#46)

ohhamma committed:
feat: game record models (#56)

ohhamma committed:
chore: delete unused files

ohhamma committed:
feat: remove datatime now option (#56)

ohhamma committed:
feat: get 1v1 game records (#56)

ohhamma committed:
fix: unintended query param problem

ohhamma committed:
feat: add 1v1 game record (#56)

ohhamma committed:
chore: change language code

ohhamma committed:
feat: get tournament list (#56)

ohhamma committed:
feat: create tournament (#56)

ohhamma committed:
fix: create duplicate game error (#60)

ohhamma committed:
feat: get tournament detail (#56)

ohhamma committed:
refactor: change tournament POST input structure (#56)

ohhamma committed:
feat: get tournament by id (#56)

ohhamma committed:
Merge branch 'develop' into feat/game

hyemch reviewed (changes requested):
postman으로 테스트 여러번 해본 결과
- `1v1s` 전적 조회 
- `1v1s` 전적 추가 
- `tournaments` 전적 조회
- `tournaments` 세부 내용 조회 
정상적으로 작동 잘 됩니다! 

다만, `tournaments` 전적 추가에서 테스트 하던 중
- `game/me/tournaments/`에서 다른 유저가 추가한 전적과 동일한 전적 추가시 500에러 
  - <img width="1078" alt="스크린샷 2024-04-12 오전 12 33 53" src="https://github.com/bitbit-Merry-go-round/Transcendence-BE/assets/102580205/64268e59-a26b-4f0f-8580-791afb90989f">
- `game/me/tournaments/` 에서 잘못된 json요청이 오는경우 `500` 에러
  - <img width="1080" alt="스크린샷 2024-04-12 오전 12 12 10" src="https://github.com/bitbit-Merry-go-round/Transcendence-BE/assets/102580205/ededc5d0-08d4-44bb-bcbf-3f85726dd84c">
- `me`가 아닌 `game/{username}/tournaments/`에서 전적 추가 가능
  - <img width="1085" alt="스크린샷 2024-04-12 오전 12 00 48" src="https://github.com/bitbit-Merry-go-round/Transcendence-BE/assets/102580205/09c0eeae-b60a-43bc-9f14-3fc5d9aff5f2">

이 부분 한 번 더 확인해 보는 것이 좋을 것 같습니다!🥺😊

ohhamma committed:
fix: duplicate tournament different username error (#56)

ohhamma committed:
fix: handle tournament create exceptions (#56)

- invalid request body
- invalid time format
- invalid player score
- duplicate game

ohhamma committed:
fix: handle 1v1 game create exceptions (#56)

- invalid request body
- invalid time format
- invalid player score

ohhamma commented:
확인 감사합니다!

> 1.  다른 유저가 추가한 전적과 동일한 전적 추가 시 생기는 문제는 해결했습니다.

> 2. 토너먼트 추가 시 요청 본문에 잘못된 형태로 들어오는 경우도 처리했습니다.
추가로 테스트 과정에서 생긴 여러 예외 상황들에 대해서 토너먼트와 1v1 게임 모두 처리했습니다.

> 3.  경로에 `me`가 포함되지 않은 경우 POST 요청 방지는 이미 이슈로 만들어두었습니다.
#61 

3번 사항을 제외하고 다시 테스트 부탁드립니다!

ohhamma committed:
chore: delete unused code

ohhamma committed:
chore: remove comment

hyemch reviewed (approved):
- `1v1s` 전적 조회
- `1v1s` 전적 추가
- `tournaments` 전적 조회, 세부 내용 조회
- `tournaments` 전적 추가 
관련한 모든 기능 잘 작동되는 것을 확인했습니다! 고생하셨습니다!
